packages/forms/src/UIForm/UIForm.component.js

```diff
@@ -46,11 +44,31 @@ export default class UIForm extends React.Component {
 	onChange(event, { schema, value }) {
 		const payload = {
 			formName: this.props.formName,
-			properties: this.props.properties,
+			properties: this.props.properties || {},
 			schema,
 			value,
 		};
-		this.props.onChange(event, payload);
+		payload.formData = mutateValue(payload.properties, schema.key, value);
```

Comment:
If the purpose is to have the new data eveytime (moz of not), you can add it in payload declaration just up there.
Or else, move it in the `if(moz)`